hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -565,21 +568,32 @@ public Action addUpdateActionStatus(final ActionStatusCreate c) {
         // permitted so by configuration. This is especially useful if the
         // action status feedback channel order from the device cannot be
         // guaranteed. However, if an action is closed we do not accept further
-        // close messages.
-        if (actionIsNotActiveButIntermediateFeedbackStillAllowed(actionStatus, action.isActive())) {
+        // close messages, unless it was a DOWNLOAD_ONLY action
+        if (actionIsNotActiveButIntermediateFeedbackStillAllowed(actionStatus, action.isActive())
+                && !actionIsDownloadOnlyAndTargetReportsFinished(action, actionStatus)) {
             LOG.debug("Update of actionStatus {} for action {} not possible since action not active anymore.",
                     actionStatus.getStatus(), action.getId());
             return action;
         }
         return handleAddUpdateActionStatus(actionStatus, action);
     }
 
+    private static boolean actionIsDownloadOnlyAndTargetReportsFinished(final JpaAction action,
+            final JpaActionStatus actionStatus) {
+        return isDownloadOnly(action) && FINISHED.equals(actionStatus.getStatus())
+                && !FINISHED.equals(action.getStatus());
+    }
+
     private boolean actionIsNotActiveButIntermediateFeedbackStillAllowed(final ActionStatus actionStatus,
```

Comment:
Would it make sense to to refactor this method and put the actionIsDownloadOnlyAndTargetReportsFinished logic in there? (maybe then this long comment in the addUpdateActionStatus method can be removed)
```suggestion
    private boolean isUpdatingActionStatusAllowed(Action action, ActionStatus actionStatus)
```